syntax = "proto3";

package chaosmonkey;

service ChaosMonkey {
    rpc UploadMatrix(ConnMatrix) returns (Status) {}
    rpc UpdateValue(MatValue) returns (Status) {}
    rpc GetMatrix(Empty) returns (ConnMatrix) {}
    rpc KillANode(KillANodeRequest) returns (Status) {}
}

enum StatusCode {
    OK = 0;
    ERROR = 1;
}

message Status {
    StatusCode ret = 1;
}

message ConnMatrix {
    message MatRow {
        repeated float vals = 1;
    }
    repeated MatRow rows = 1;
}

message KillANodeRequest {
    int32 node_index = 1;
}


message MatValue {
    int32 row = 1;
    int32 col = 2;
    float val = 3;
}

message Empty {}
